transports/websocket/src/fallback.rs

```diff
@@ -0,0 +1,133 @@
+// Copyright 2019 Parity Technologies (UK) Ltd.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+// OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+// DEALINGS IN THE SOFTWARE.
+
+use std::pin::Pin;
+use std::task::{Poll, Context};
+use std::io::{self, Read, IoSliceMut};
+use futures::{AsyncRead, AsyncWrite, Stream, Sink};
+use bytes5::Buf;
+use crate::framed::{TlsOrPlain, Connection, OutgoingData};
+
+#[derive(Debug)]
+pub struct Fallback<T> {
```

Comment:
`Fallback` could contain the original handshake error that caused the fall back. Then you do not need to add another error case `Error::FellBack` but you could return the more meaningful error in `WsConfig::listen_on`.

Reply:
Yep, was thinking that.